File: mobile/lib/types/setting.dart
// Mirror of `lorewyld_types::setting::Setting`.

class Setting {
  final String uuid;
  final String name;
  final String? descriptionNoteUuid;
  final String ownerUserUuid;
  final String? publishedAsModuleUuid;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Setting({
    required this.uuid,
    required this.name,
    this.descriptionNoteUuid,
    required this.ownerUserUuid,
    this.publishedAsModuleUuid,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Setting.fromJson(Map<String, dynamic> json) => Setting(
        uuid: json['uuid'] as String,
        name: json['name'] as String,
        descriptionNoteUuid: json['description_note_uuid'] as String?,
        ownerUserUuid: json['owner_user_uuid'] as String,
        publishedAsModuleUuid: json['published_as_module_uuid'] as String?,
        createdAt: DateTime.parse(json['created_at'] as String),
        updatedAt: DateTime.parse(json['updated_at'] as String),
      );
}

File: shared/types/src/setting.rs
use serde::{Deserialize, Serialize};

use crate::common::{EntityId, Timestamp};

/// A user-owned worldbuilding workspace — the **promotion staging
/// ground** in the platform's three-tier content commitment ramp:
///
///   campaign-scoped *(tonight's game)*
///     → setting-scoped *(my reusable world)*
///     → module-scoped *(published, shareable package)*
///
/// Settings hold lore notes (`LoreNote` with `NoteScope::Setting`) and,
/// in later tiers, structured homebrew records the user is developing
/// for a setting. The Promote-to-Module wizard takes a Setting and
/// produces a publishable `ContentModule` via snapshot publication.
/// After publication, the source Setting persists with
/// `published_as_module_uuid` set, and continues to evolve
/// independently of the frozen module versions it has spawned.
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct Setting {
    pub uuid: EntityId,
    pub name: String,
    /// Optional `SettingScope` lore note serving as the world primer.
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub description_note_uuid: Option<EntityId>,
    /// `None` when the owning account has been deleted — content
    /// outlives its owner.
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub owner_user_uuid: Option<EntityId>,
    /// Set when the setting has been published as a module via the
    /// Promote-to-Module wizard. Points to the most recent published
    /// version; older versions are reachable via the module's own
    /// `previous_version_uuid` chain.
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub published_as_module_uuid: Option<EntityId>,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}

/// Co-author relationship — a user invited to collaborate on a setting
/// has read/write access to its notes and (in later tiers) its draft
/// structured records.
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct SettingCollaborator {
    pub setting_uuid: EntityId,
    pub user_uuid: EntityId,
}
